query/unparsed_query.go

```diff
@@ -318,6 +319,9 @@ func (q *UnparsedQuery) validateGroupBy() error {
 	if q.TimeGranularity != nil {
 		return errTimeGranularityWithGroupBy
 	}
+	if len(q.Calculations) == 0 {
```

Comment:
Perhaps move this to `parseCalculations`? Tend to keep this validation logic more manageable by only validating fields above not below.